### PR TITLE
networkctl: Add support to display macvlan/macvtap mode

### PR DESCRIPTION
**Pull request by ssahani**
cc @yuwata

### CODE REVIEW

**Comment by yuwata** [comment]
LGTM.

File: src/network/networkctl.c [hunk 1]
```diff
@@ -1611,6 +1615,25 @@ static int link_status_one(
...
+                        [ MACVLAN_MODE_VEPA ]     = { "vepa" },
+                        [ MACVLAN_MODE_BRIDGE ]   = { "bridge" },
+                        [ MACVLAN_MODE_PASSTHRU ] = { "passthru" },
+                        [ MACVLAN_MODE_SOURCE ]   = { "source" },
+                };
+
+                r = table_add_many(table,
+                                   TABLE_EMPTY,
+                                   TABLE_STRING, "Mode:",
+                                   TABLE_STRING, mode_table[info->macvlan_mode]);
```
Comment (keszybz) [inline src/network/networkctl.c]: Hmm, no, that is rather dangerous. We cannot trust the data send from the remote side blindly. Apart from bugs, the remote might get support for newer macvlan modes and then we'd just crash here. I'd very much prefer to have those tables in a separate file (to not clutter the networkctl.c file too much), and go through a proper accessor function.

**Comment by ssahani** [comment]
updated thanks for the review @keszybz . PTAL

File: src/shared/macvlan-util.c [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <net/if.h>
+
+#include "conf-parser.h"
```
Comment (yuwata) [inline src/shared/macvlan-util.c:3]: The header is not necessary.

Comment (ssahani) [reply src/shared/macvlan-util.c:3]: dropped. Thanks

File: src/network/netdev/macvlan.c [hunk 3]
```diff
@@ -4,16 +4,9 @@
 
 #include "conf-parser.h"
 #include "macvlan.h"
+#include "macvlan-util.h"
 #include "string-table.h"
```
Comment (yuwata) [inline src/network/netdev/macvlan.c]: Please also drop `string-table.h` here.

Comment (ssahani) [reply src/network/netdev/macvlan.c]: dropped thanks .

**Comment by yuwata** [comment]
Same here. Please drop them in the first commit.

File: src/network/netdev/macvlan.c (hunk 3 above)
Comment (yuwata) [reply src/network/netdev/macvlan.c]: Not dropped...

File: src/shared/macvlan-util.c (hunk 2 above)
Comment (yuwata) [reply src/shared/macvlan-util.c:3]: Not dropped...

File: src/network/netdev/macvlan.c (hunk 3 above)
Comment (ssahani) [reply src/network/netdev/macvlan.c]: ohh now dropped. I missed

**Comment by poettering** [comment]
Merged as cf217a0922571ec3705eb8d8818723b0ec68846f

**Pull request closed without merging**